### ISSUE TITLE
check_balancer_running method never returns True on 3.4

### ISSUE THREAD

**Issue by acheccuc** [Percona-Lab/mongodb_consistent_backup#136]
We have a small MongoDB 3.4 sharded cluster where we have been testing MongoDB Consistent Backup.
After configuring the YML conf file we have been able to start the operation, but we only got this repeated message before a time-out kicked in:

[2017-05-12 13:59:44,684] [INFO] [MainProcess] [Sharding:stop_balancer:113] Balancer is still running, sleeping for 3 sec(s)

Having a look into the code, it seems like check_balancer_running() never returns True because config.locks.state is always 2 in 3.4, the MongoDB documentation reports:

> Changed in version 3.4: As of version 3.4, the state field will always have a value 2 to prevent any legacy mongos instances from performing the balancing operation. The when field specifies the time when the config server member became the primary.

While check_balancer_running() looks for 0 to ensure that the balancer is disabled.

```
def check_balancer_running(self):
        try:
            config = self.connection['config']
            lock   = config['locks'].find_one({'_id': 'balancer'})
            if 'state' in lock and int(lock['state']) == 0:
                return False
            return True
        except Exception, e:
raise DBOperationError(e)
```

Could you please have a look?

**Comment by timvaillancourt** [comment]
Thanks @acheccuc. I've noticed the same problem and had not looked into why yet. It looks like the code just needs to support the new lock value.

This will be added to 1.0.2 or 1.0.3 soon.

**Comment by acheccuc** [comment]
Thanks @timvaillancourt , very appreciated

**Comment by timvaillancourt** [comment]
Hello @acheccuc. This was merged to our 1.0.2 branch.

You can try this here: https://github.com/Percona-Lab/mongodb_consistent_backup/tree/1.0.2

or 

```
git clone -b 1.0.2 https://github.com/Percona-Lab/mongodb_consistent_backup
```

Currently you need to use a 'mongos' for this code to work. This will be merge to 'master' soon.
